fhir-server/src/main/java/com/ibm/fhir/server/util/FHIRRestHelper.java

```diff
@@ -2330,7 +2330,19 @@ private String getAbsoluteUri(String baseUri, String relativeUri) {
             if (HTTPReturnPreference.REPRESENTATION.equals(FHIRRequestContext.get().getReturnPreference())) {
                 bundleEntryBuilder.resource(resource);
             } else if (HTTPReturnPreference.OPERATION_OUTCOME.equals(FHIRRequestContext.get().getReturnPreference())) {
-                bundleEntryBuilder.resource(operationOutcome);
+                OperationOutcome responseEntryOutcome = (OperationOutcome)responseEntry.getResource();
+                if (responseEntryOutcome != null && !responseEntryOutcome.equals(FHIRUtil.ALL_OK)) {
+                    if (operationOutcome.equals(FHIRUtil.ALL_OK)) {
+                        bundleEntryBuilder.resource(responseEntryOutcome);
```

Comment:
Does it matter whether the request was a `batch` vs a `transaction`?
I know for failed `transaction` requests we send back a single OperationOutcome (vs. sending one-per-resource in `batch`)...just want to make sure this logic works for both bundle types.

Reply:
Yes, this works for both types. This method will only get called after a successful call to the persistence layer, so the only change is that on a successful request, the OperationOutcome object may now contain warnings that it didn't contain before. In the case of a failed `transaction` request during validation, that code path is unchanged - it returns all validation warnings/errors for the failed bundle entry only. In the case of a failed `transaction` request during create or update, that code path is also unchanged from before. An exception is thrown and a bad request code is returned.